### PR TITLE
Tests added for follower fetch protocol (part-1)

### PR DESCRIPTION
**Pull request by kamalcph**
Tests added/updated for the below classes:

1. LogManagerTest
2. ProducerStateManagerTest
3. UnifiedLogTest
4. RemoteIndexCacheTest
5. LeaderEpochFileCacheTest

Opensource PR: https://github.com/apache/kafka/pull/11390

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/log/LogManager.scala [hunk 1]
```diff
@@ -393,10 +393,8 @@ class LogManager(logDirs: Seq[File],
 
         val logsToLoad = Option(dir.listFiles).getOrElse(Array.empty).filter(logDir =>
           logDir.isDirectory &&
-            UnifiedLog.parseTopicPartitionName(logDir).topic != KafkaRaftServer.MetadataTopic &&
-            // Ignore remote-log-index-cache directory as that is index cache maintained by tiered storage subsystem
-            // but not any topic-partition dir.
-            !logDir.getName.equals(RemoteIndexCache.DirName))
+            !logDir.getName.equals(RemoteIndexCache.DirName) &&
```
Comment (satishd) [inline core/src/main/scala/kafka/log/LogManager.scala:396]: Why are they shuffled and removed the javadoc?

File: core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala [hunk 2]
```diff
@@ -16,105 +16,94 @@
...
+    val txnIdxFile = new File(tpDir, "txn-index" + UnifiedLog.TxnIndexFileSuffix)
+    txnIdxFile.createNewFile()
+    when(rsm.fetchIndex(any(classOf[RemoteLogSegmentMetadata]), any(classOf[IndexType])))
+      .thenAnswer(ans => {
+        val metadata = ans.getArgument[RemoteLogSegmentMetadata](0)
+        val indexType = ans.getArgument[IndexType](1)
+        val maxEntries = (metadata.endOffset() - metadata.startOffset()).asInstanceOf[Int]
+        val offsetIdx = new OffsetIndex(new File(tpDir, String.valueOf(metadata.startOffset()) + UnifiedLog.IndexFileSuffix),
+          metadata.startOffset(), maxIndexSize = maxEntries * 8)
+        val timeIdx = new TimeIndex(new File(tpDir, String.valueOf(metadata.startOffset()) + UnifiedLog.TimeIndexFileSuffix),
```
Comment (satishd) [inline core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala:66]: Why do we need to create these index files always even when the indextype is not meant for these?

File: core/src/main/scala/kafka/log/LogManager.scala (hunk 1 above)
Comment (kamalcph) [reply core/src/main/scala/kafka/log/LogManager.scala:396]: Without this change, the newly added test in  LogManagerTest#testLoadLogsSkipRemoteIndexCache will fail. As UnifiedLog.parseTopicPartitionName(logDir) throws error if the directory is not in format (topic-partition). The javadoc seems redundant to me, as the code already conveys the meaning.

File: core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala (hunk 2 above)
Comment (kamalcph) [reply core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala:66]: To append the index entries for offsetIdx and timeIdx. The entries are appended only once for each test.

File: core/src/main/scala/kafka/log/LogManager.scala (hunk 1 above)
Comment (satishd) [reply core/src/main/scala/kafka/log/LogManager.scala:396]: Please leave the javadoc message as the readers may not know tiered storage parts.

**Review by satishd — APPROVED** [review]
Thanks @kamalcph for the PR, LGTM.

**Pull request merged**